source/standards/diagrams-as-code.html.md.erb

```diff
@@ -0,0 +1,40 @@
+---
+title: Diagrams as code
+last_reviewed_on: 2024-11-13
+review_in: 6 months
+---
+
+# <%= current_page.data.title %>
+
+Code can be used to define the content, structure and appearance of a diagram which can then be rendered, instead of using a drawing tool directly. This approach is known as "diagrams as code".
+
+## What are diagrams for?
+
+> A good diagram enables you to more easily convey your design to others, and it provides a focus for your creative, as well as your technical thinking.
+— [Drawing good architecture diagrams](https://www.ncsc.gov.uk/blog-post/drawing-good-architecture-diagrams) tips from NCSC
+
+There are many ways to create a diagram, from pen and paper or whiteboard through to specialist diagramming software. The right tool will depend on the circumstances, including what the diagram is for and who needs to be able to contribute to it, but it's always worth considering a diagram generated from code.
+
+Diagrams are a form of documentation, and some teams already use a '[docs as code](https://technology.blog.gov.uk/2017/08/25/why-we-use-a-docs-as-code-approach-for-technical-documentation/)' approach. Diagrams as code builds on the rationale and benefits of docs as code, including
+
+* keeping documentation close to source code
+* ease of change
+* a workflow to review changes
+* version history
+* consistency
+
+## Diagrams as code tools
```

Comment:
Appreciate this is only an example, but perhaps worth calling out a couple of other tools? C4 perhaps? Others that are in use?

Reply:
Notify have been using D2, and also mermaid in markdown depending on the diagram complexity and team member preference.  As mentioned on the last Technical Architects Fortnightly, it is good to suggest but not mandate the tool.

Reply:
This was discussed at the GDS Way Forum on 2024-11-27

I've added Mermaid sequence diagrams as that's the tool and diagram type I have the most experience using. The intention is that after this initial PR is merged and the page is created, others can add details about tools they have successfully used, ideally with links to examples. [C4](https://c4model.com/) is a tooling-independent model, and I'm very interested in exploring the use of [Structurizr](https://www.structurizr.com/) - I will of course add it here if successful...